### ISSUE TITLE
No menu entry to add shortcut to home screen

### ISSUE THREAD

**Issue by opk12** [vector-im/element-android#4423]
### Steps to reproduce

On my phone, the room settings screen and the app's long-tap context menu have no option to add a shortcut on the home screen. Shortcuts do work in other apps on the same phone.

![element_bug_shortcut](https://user-images.githubusercontent.com/40740650/140608316-74406510-2b7f-4477-80d3-8584fac3a5ac.png)

#### What did you expect?
To see options to add a shortcut to the home.

#### What happened instead?
There are no options to add a shortcut to the home.

### Your phone model

Samsung

### Operating system version

Android 7

### Application version and app store

Element Android 1.3.3 (40103030) from F-Droid

### Homeserver

matrix.org

### Will you send logs?

No

**Comment by bmarty** [comment]
The app is checking [here](https://github.com/vector-im/element-android/blob/main/vector/src/main/java/im/vector/app/features/roomprofile/RoomProfileController.kt#L170) the returned value of [isRequestPinShortcutSupported()](https://developer.android.com/reference/android/content/pm/ShortcutManager#isRequestPinShortcutSupported()) before displaying this action item.
There is maybe a missing permission, can you check on the device app settings?

**Comment by opk12** [comment]
`Install shortcuts` and `uninstall shortcuts` are in the list for other apps, but not for Element. Do I report to F-droid?

![tutti](https://user-images.githubusercontent.com/40740650/140759103-d6f4605f-7998-4d8d-bec4-ad617f71251d.png)
![all1](https://user-images.githubusercontent.com/40740650/140759155-ea84691c-c2e3-4dbb-abdb-882bb409f5fa.png)
![all2](https://user-images.githubusercontent.com/40740650/140759176-050ec59c-316f-4ff0-996f-2a3476f98437.png)

**Comment by ouchadam** [comment]
which version of android 7 is the device running? from my understanding android added shortcut support in 7.1.1, so if this device is running 7.0 then unfortunately we won't be able to add shortcuts

https://developer.android.com/reference/android/content/pm/ShortcutManager.html (Api 25)
https://developer.android.com/guide/topics/manifest/uses-sdk-element#ApiLevels

**Comment by opk12** [comment]
Version 7.1.1. As said above, the other apps have working shortcuts and have `install shortcuts`/`remove shortcuts` in the "All permissions" list, Element does not.

**Comment by ouchadam** [comment]
@bmarty this might actually be a simple case of using the `ShorcutManagerCompat` instead

By default the system version seems to support android 8.0+ but the compat version can go lower

> Note: See also the support library counterpart ShortcutManagerCompat.isRequestPinShortcutSupported(Context), which supports Android versions lower than VERSION_CODES#O using the legacy private intent com.android.launcher.action.INSTALL_SHORTCUT.

**Comment by bmarty** [comment]
Yes, this is the current imple : https://github.com/vector-im/element-android/blob/main/vector/src/main/java/im/vector/app/features/home/ShortcutCreator.kt#L57. Is there something else you are thinking about?

**Comment by ouchadam** [comment]
ah :facepalm: I confused the `ShortcutCreator` for the system version

**Comment by opk12** [comment]
The `X-Needs-Info` label is still present - is there other info I can provide about the phone?

**Comment by ouchadam** [comment]
nope, only needed to confirm the exact android 7 version, have removed

**Comment by opk12** [comment]
Ok. I have not experience in Android development, but where do the `install shortcuts` / `remove shortcuts` in the "All permissions" list come from for the other apps? I mean, the app has a harc-coded manifest with the permissions list? Or a build tool generates the list based on the invoked system library's methods?

**Comment by ouchadam** [comment]
ideally the permission should come through via the `ShortcutCompat` library but there's something strange going on...

I'm also able to reproduce the issue on a 7.1.1 emulator and fix it by manually adding the `com.android.launcher.permission.INSTALL_SHORTCUT` permission

| 7.1.1 | 10 | 7.1.1 WITH PERMISSION | 7.1.1 LAUNCHER (WP)
| --- | --- | --- | --- |
![2021-11-17T17:53:47,088292286+00:00](https://user-images.githubusercontent.com/1848238/142255549-a55cbfae-0207-4bec-9e15-9439fe780ce8.png)|![Screenshot_20211117_175523](https://user-images.githubusercontent.com/1848238/142255674-762bf0e7-8abc-4e8d-af7a-f6f3003e720e.png)|![2021-11-17T17:59:18,282005910+00:00](https://user-images.githubusercontent.com/1848238/142256937-598976e6-644b-4c8d-b8ca-7d2cc3f8ac38.png)|![2021-11-17T18:03:21,888363673+00:00](https://user-images.githubusercontent.com/1848238/142256945-c97da204-8d79-49ea-b236-5daeb9ab8e97.png)


For some reason the shortcut permission is being commented by the [ShortcutBadger library](https://github.com/leolin310148/ShortcutBadger/blob/master/ShortcutBadger/src/main/AndroidManifest.xml#L9) out during the manifest merging process 

![2021-11-17T18:05:37,932070832+00:00](https://user-images.githubusercontent.com/1848238/142257096-cc75c6da-71c8-4541-8993-63b6b62955f7.png)

**Comment by ouchadam** [comment]
it seems the workaround is to long press the element icon from the launcher which will show shortcut options and then long press those to add to the homescreen

**Comment by opk12** [comment]
> it seems the workaround is to long press the element icon from the launcher which will show shortcut options and then long press those to add to the homescreen

How? When I long-press the Element icon on the home (Samsung builtin home screen), I have `Select multiple items` - `Remove shortcut` - `Sleep` - `Uninstall` - `App info`. Long-pressing them has the effect of a short press.

**Comment by bmarty** [comment]
> For some reason the shortcut permission is being commented by the ShortcutBadger library out during the manifest merging process

It was commented 7 years ago...

Maybe just add

```xml
<uses-permission
        android:name="com.android.launcher.permission.INSTALL_SHORTCUT"
        android:maxSdkVersion="25" />
```

in our Manifest as a workaround, WDYT @ouchadam ?

**Comment by ouchadam** [comment]
> How? When I long-press the Element icon on the home (Samsung builtin home screen), I have `Select multiple items` - `Remove shortcut` - `Sleep` - `Uninstall` - `App info`. Long-pressing them has the effect of a short press.

I see, this must be launcher dependant, using the Pixel launcher on 7.1.1 I'm able to long press the launcher icon which reveals   the launcher shortcuts which I can then long press to add to the 

-----

> ```
> <uses-permission
>         android:name="com.android.launcher.permission.INSTALL_SHORTCUT"
>         android:maxSdkVersion="25" />
> ```
> 
> in our Manifest as a workaround, WDYT @ouchadam ?

sounds reasonable to me
